woocommerce/woocommerce-ios
# Shipping settings: Comma as decimal separator.

Closes: #6567

### Description
While editing a product in iOS, if we use a comma as a decimal separator for shipping package dimensions, the decimal point is not saved in the backend, upon saving the product. The backend ignores the comma and saves the number.

This PR attempts to solve the issue by,

- Always convert the weight and package dimensions to the device's locale before displaying. 
- The weight and package dimensions are converted back to the US locale before sending the request to API. This ensures that the API saves the decimal point information.

More details about this approach at p2 post - p91TBi-8kO-p2

### Testing instructions
1. Change the device's region to a region that uses commas as decimal points. (Vietnam for example)
1. Navigate to the Products tab.
1. Select the "+" button and choose the physical product type
1. Fill in basic details for the product: title, price.
1. Select "Add more details" > "Shipping". Enter decimal values for the weight and dimensions and save changes.
1. Save or publish the product. Notice that the weight and dimensions are displayed with proper decimal points, using comma as a decimal separator.

### Screenshots
![CommaDecimalSeparator](https://user-images.githubusercontent.com/524475/165274663-67a515dd-3901-455c-b16d-a0a3976312da.gif)


---
